gflownet/envs/crystals/crystal.py

```diff
@@ -535,3 +541,242 @@ def readable2state(self, readable: str) -> List[int]:
+            color=colors["full"],
+            density=True,
+        )
+        ax[1].axvline(
+            mean_r,
+            color=colors["full"],
+            linestyle=(0, (5, 10)),
+            label=f"Mean = {mean_r:.3f}",
+        )
+        ax[1].axvline(
+            mean_r + std_r,
+            color=colors["full"],
+            linestyle=(0, (1, 10)),
+            label=f"Std = {std_r:.3f}",
+        )
+        ax[1].axvline(
+            mean_r - std_r,
+            color=colors["full"],
+            linestyle=(0, (1, 10)),
+        )
+        ax[1].hist(
+            top_k_r,
+            bins=100,
+            alpha=0.7,
+            label=f"Top k = {top_k}",
+            color=colors["top_k"],
+            density=True,
+        )
+        ax[1].axvline(
+            mean_top_k_r,
+            color=colors["top_k"],
+            linestyle=(0, (5, 10)),
+            label=f"Mean = {mean_top_k_r:.3f}",
+        )
+        ax[1].axvline(
+            mean_top_k_r + std_top_k_r,
+            color=colors["top_k"],
+            linestyle=(0, (1, 10)),
+            label=f"Std = {std_top_k_r:.3f}",
+        )
+        ax[1].axvline(
+            mean_top_k_r - std_top_k_r,
+            color=colors["top_k"],
+            linestyle=(0, (1, 10)),
+        )
+        ax[1].set_title(
+            f"Reward distribution for {top_k} vs {len(reward)}"
+            + f" samples\nBest: {best_r:.3f}",
+            y=0,
+            pad=-20,
+            verticalalignment="top",
+            size=12,
+        )
+        ax[1].legend()
+        title = f"{name.capitalize()} energy and reward distributions"
+        if step is not None:
+            title += f" (step {step})"
+        fig.suptitle(title, y=0.95)
+        plt.tight_layout(rect=[0, 0.02, 1, 0.98])
+
+        metrics = {
+            f"Mean {name} energy": mean_e,
+            f"Std {name} energy": std_e,
+            f"Mean {name} reward": mean_r,
+            f"Std {name} reward": std_r,
+            f"Mean {name} top k energy": mean_top_k_e,
+            f"Std {name} top k energy": std_top_k_e,
+            f"Mean {name} top k reward": mean_top_k_r,
+            f"Std {name} top k reward": std_top_k_r,
+            f"Best (min) {name} energy": best_e,
+            f"Best (max) {name} reward": best_r,
+        }
+        figs = [fig]
+        fig_names = [title]
+
+        if name.lower() == "train ground truth":
+            proxy_metrics, proxy_figs, proxy_fig_names = self.top_k_metrics_and_plots(
+                None,
+                top_k,
+                "train proxy",
+                energy=proxy,
+                reward=proxy_reward,
+                step=None,
+                **kwargs,
+            )
+            metrics.update(proxy_metrics)
+            figs += proxy_figs
+            fig_names += proxy_fig_names
+
+        return metrics, figs, fig_names
+
+    @torch.no_grad()
+    def compute_train_energy_proxy_and_rewards(self):
```

Comment:
This looks more like a proxy method than a env method, right? Actually, this would potentially break if the proxy changes. Also, I could use a brief docstring here.

Reply:
Agreed